gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java

```diff
@@ -387,15 +387,15 @@ private void setRecursivePermission(Path path, OwnerAndPermission ownerAndPermis
     Collections.reverse(files);
 
     for (FileStatus file : files) {
-      safeSetPathPermission(file.getPath(), addExecutePermissionsIfRequired(file, ownerAndPermission));
+      safeSetPathPermission(this.fs, file.getPath(), addExecutePermissionsIfRequired(file, ownerAndPermission));
     }
   }
 
   /**
    * The method makes sure it always grants execute permissions for an owner if the <code>file</code> passed is a
    * directory. The publisher needs it to publish it to the final directory and list files under this directory.
    */
-  private static OwnerAndPermission addExecutePermissionsIfRequired(FileStatus file,
+  public static OwnerAndPermission addExecutePermissionsIfRequired(FileStatus file,
```

Comment:
while you're at it--and before it becomes `public`--I would rename to `setOwnerExecuteBitIfDirectory`... 
(should be safe, as this was previously `private`)